### DISCUSSION TITLE
next/image, how to set image height and width to `100%` ?

### DISCUSSION THREAD

**Discussion by AyushAgnihotri2025** [vercel/next.js#44332] [Help] [answered]
I have tried to define my image like this:

```html
<Image src={props.image.url} alt="Image" height='100%' width='100%'/>
```

which is showing this error:

`Type '"100%"' is not assignable to type 'SafeNumber | undefined'.`

Note : I am using `13.1.1` version of Next.js

**Comment by muhammadjufry** [comment]
You can do this to set your width image to 100% full:

```
<Image
  src="/example.png"
  layout="fill"
  sizes="100vw"
/>
```

Hope this work for you!

**Comment by ciruz** [comment, marked as answer]
With next.js ``13.x`` you can now simply use the ``fill`` attribute, like so:

```javascript
<Image src={image} alt="" fill />
```

result:
![image](https://user-images.githubusercontent.com/987947/209806162-69db51a9-6229-44fb-8631-4301007fcfb0.png)


**If you want to style it, the ``parent`` element must have ``position: relative`` and ``display: block``.**

If you want to control the size of the image, you use the parent element with fill, like so:
```javascript
<div
  style={{
    position: "relative",
    width: "300px",
    height: "100px",
    border: "solid 3px blue",
  }}
>
    <Image src={image} alt="" fill />
</div>
```

result:
![image](https://user-images.githubusercontent.com/987947/209806616-9df49be7-87f0-4de6-b553-81e62ab6cb4d.png)


Or if this is not the desired look, use ``object-fit`` now via ``css/style`` instead of an attribute, like it was the case with next 12.

```javascript
<div
  style={{
    position: "relative",
    width: "300px",
    height: "100px",
    border: "solid 3px blue",
  }}
>
    <Image src={image} alt="" fill style={{ objectFit: "cover" }} />
</div>
```

result:
![image](https://user-images.githubusercontent.com/987947/209806834-aea14a83-07ae-4833-bda6-e6e022fff0db.png)

**Reply by AyushAgnihotri2025** [reply]
Thanks sir, You saved my project.
